File: ast/types/type_name.go
package types

import (
	"github.com/panda-io/micro-panda/ast/core"
)

type TypeName struct {
	TypeBase
	Name      string
	Qualified string

	IsEnum bool
}

func (n *TypeName) Equal(t core.Type) bool {
	if t, ok := t.(*TypeName); ok {
		return n.Qualified != "" && t.Qualified == n.Qualified
	}
	return false
}

func (n *TypeName) String() string {
	if n.Qualified != "" {
		return n.Qualified
	}
	return n.Name
}

File: app/src/main/java/com/github/panda_io/micro_panda/ast/type/TypeName.java
package com.github.panda_io.micro_panda.ast.type;

public class TypeName extends Type {
    public String name;
    public String qualified;
    public boolean isEnum;

    public TypeName() {
        this.isEnum = false;
    }

    public boolean equal(Type type) {
        if (type instanceof TypeName) {
            return this.qualified != null && this.qualified.equals(((TypeName) type).qualified);
        }
        return false;
    }

    public String string() {
        if (this.qualified != null) {
            return this.qualified;
        }
        return this.name;
    }
}
